python-package/lightgbm/arrow.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+"""Utilities for handling Arrow in LightGBM."""
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import Iterator, Union
+
+import pyarrow as pa
+from pyarrow.cffi import ffi
+
```

Comment:
```suggestion

__all__: List[str] = []
```

Please make it clear that nothing iin this module is intended to be part of the public API, e.g. that:

* it won't have dedicated entries in the docs generated with `sphinx` autodoc
* it won't be pulled in by `from lightgbm import *`